third_party/move/move-compiler-v2/src/unused_params_checker.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implements an environment pipeline which checks for unused type parameters in struct definitions.
```

Comment:
Would be good to add if there are any pre-conditions (such as passes and checks that this assumes), so that we can re-order stuff.

Eg, we may require recursive struct checks to be already done, otherwise, we may have infinite recursion here.

Reply:
done